components/fonts/font_cache_thread.rs

```diff
@@ -417,52 +448,33 @@ impl From<&FontFaceRuleData> for CSSFontFaceDescriptors {
     }
 }
 
-impl FontCacheThread {
-    pub fn new(webrender_api: Box<dyn WebRenderFontApi + Send>) -> FontCacheThread {
-        let (chan, port) = ipc::channel().unwrap();
-
-        thread::Builder::new()
-            .name("FontCache".to_owned())
-            .spawn(move || {
-                #[allow(clippy::default_constructed_unit_structs)]
-                let mut cache = FontCache {
-                    port,
-                    font_data: HashMap::new(),
-                    local_families: Default::default(),
-                    webrender_api,
-                    webrender_fonts: HashMap::new(),
-                    font_instances: HashMap::new(),
-                    generic_fonts: Default::default(),
-                };
-
-                cache.refresh_local_families();
-                cache.run();
-            })
-            .expect("Thread spawning failed");
-
-        FontCacheThread { chan }
-    }
-
+impl SystemFontServiceProxy {
     pub fn exit(&self) {
         let (response_chan, response_port) = ipc::channel().unwrap();
-        self.chan
+        self.sender
+            .lock()
```

Comment:
What are the other threads with access to a sender? Only script right?

I think instead of the ping mechanism, we should ensure that the threads other than the constellation(which is the only calling `exit` right?) have already been shut-down before, which I think is actually the case already.

Reply:
Actually, depends on https://github.com/servo/servo/issues/30849

Reply:
> What are the other threads with access to a sender? Only script right?

I think that canvas threads also has access to this now, but probably that will need to be proxied through the `Document` / script as well ultimately.

> I think instead of the ping mechanism, we should ensure that the threads other than the constellation(which is the only calling `exit` right?) have already been shut-down before, which I think is actually the case already.

This seems like a good goal once #30849 is taken care of and the change above is made.